alacritty/src/event.rs

```diff
@@ -94,6 +95,9 @@ pub struct SearchState {
 
     /// Search origin in viewport coordinates relative to original display offset.
     origin: Point,
+
+    /// Focused match during vi-less search.
+    focused_match: RangeInclusive<Point<usize>>,
```

Comment:
Could we make it `Option<RangeInclusive<Point<usize>>>` , since the logic `self.search_state.focused_match != (Point::default()..=Point::default())` looks just bad to me, and we already have a bug from that logic where you could have top left cell highlighted where it shouldn't? Just error prone tbqh.

It's also more logical to have `Option`, since if you have search active, but haven't typed anything you won't have any match, however right now I have an active match for normal search if I don't even type on the bottom left corner. To repro requires having text in the bottom left corner of the terminal(not viewport) and scrolling down one line, so you can see it.

![picture](https://user-images.githubusercontent.com/27620401/98619802-03cd3e80-2315-11eb-980a-77fa28dddcf8.png)

Reply:
> we already have a bug from that logic where you could have top left cell highlighted where it shouldn't?

I'm not sure that's related to this Default..=Default is not a range that includes anything.

But sure, I'll look into it.